### PR TITLE
Adding PythonFileToolSpec

### PR DESCRIPTION
**Pull request by ajhofmann**
Extracting the Python file tool spec from https://github.com/emptycrown/llama-hub/blob/main/llama_hub/tools/notebooks/create_a_tool.ipynb as a separate tool

### CODE REVIEW

File: llama_hub/tools/python_file/__init__.py [hunk 1]
```diff
@@ -0,0 +1 @@
+# __init__.py
```
Comment (jerryjliu) [inline llama_hub/tools/python_file/__init__.py:1]: nit: unused?

**Review by jerryjliu — APPROVED** [review]

File: llama_hub/tools/python_file/README.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+This tool loads a python file and extracts function names, arguments and descriptions automatically. This tool is particular useful for automatically creating custom Tool Specs when you already have well documented python functions. 
+
+## Usage
+
+This tool has more extensive example usage documented in a Jupyter notebook [here](https://github.com/emptycrown/llama-hub/tree/main/llama_hub/tools/notebooks/create_a_tool.ipynb)
+
+Here's an example usage of the PythonFileToolSpec.
+
+```python
+from llama_hub.tools.openapi.base import PythonFileToolSpec
```
Comment (jerryjliu) [inline llama_hub/tools/python_file/README.md:12]: import is wrong

**Review by jerryjliu — APPROVED** [review]
one nit, but approving to unblock

File: llama_hub/tools/python_file/README.md (hunk 2 above)
Comment (ajhofmann) [reply llama_hub/tools/python_file/README.md]: not sure If I had too much or not enough coffee yesterday 😮‍💨

**Pull request merged**